### PR TITLE
ChallengingTask

### PR DESCRIPTION
**Pull request by MatviiGD**

### CODE REVIEW

File: src/ChallengingTaskCarSharing/Database.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package ChallengingTaskCarSharing;
+
+import java.sql.*;
+
+
+public class Database {
+    public Connection connection;
```
Comment (rzherebgrid) [inline src/ChallengingTaskCarSharing/Database.java]: private

File: src/ChallengingTaskCarSharing/MenuService.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+    public MenuService(CarDAO carDAO, CompanyDAO companyDAO, CustomerDAO customerDAO, MenuView view) {
+        this.sc = new Scanner(System.in).useDelimiter("\n");
+        this.carDAO = carDAO;
+        this.companyDAO = companyDAO;
+        this.customerDAO = customerDAO;
+        this.view = view;
+    }
+
+    void addCompany() {
```
Comment (rzherebgrid) [inline src/ChallengingTaskCarSharing/MenuService.java]: public?

File: src/ChallengingTaskCarSharing/MenuService.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        if (customer.getRentedCar() == null) {
+            var companies = companyDAO.getAllCompanies();
+            if (!companies.isEmpty()) {
+                System.out.printf("\nChoose a company:\n%s\n0. Back\n", getAsIndexedList(companies));
+                var company = chooseElementFromList(companies);
+                if (company != null) {
+                    var cars = carDAO.getAllAvailableCompanyCars();
+                    if (!cars.isEmpty()) {
+                        System.out.printf("\nChoose a car:\n%s\n0. Back\n", getAsIndexedList(cars));
+                        var car = chooseElementFromList(cars);
```
Comment (rzherebgrid) [inline src/ChallengingTaskCarSharing/MenuService.java]: Change var to Car, it's not good looking  to have type casting bellow

**Review by rzherebgrid — CHANGES_REQUESTED** [review]
1. Every method should have a modifier
2. A lot of If-else blocks without brackets

**Review by rzherebgrid — APPROVED** [review]

**Comment by MatviiGD** [comment]
thank you

**Pull request merged**